Changes in the logic of creating tags

serhii-nochevnyi opened the pull request:
Removed posibility to create a tag as empty string or a string containing only spaces.

Commit:
It's removes possibility to create a tag as empty string or a string containing only spaces

Slatch commented on src/qless-core/qless.lua:
```diff
@@ -29,6 +29,25 @@ function table.extend(self, other)
+
+local tagUtils = {}
+
+function tagUtils.isEmpty(tag)
```
I think it is possible not to bind to the word `tag` (in `tagUtils`), because the function checks all strings for emptiness, not just tags.

serhii-nochevnyi replied:
Agree. I think the name of the function needs to be changed as well

Slatch commented on tests/Tags/TagsTest.php:
```diff
@@ -0,0 +1,55 @@
+        $queue->put('Foo', [], 'jid-13', 0, 0, 0, ['', 'tag-test5', 'tag-test6', ' ']);
+        $queue->pop()->tag('', ' ', '   ', ' tag-test8');
+
+        self::assertEquals(['tag-test5', 'tag-test6', ' tag-test8'], $this->client->jobs['jid-13']->tags);
```
Controversial point with spaces in tags. I think they should be trimmed too. `     test-tag   ` (with the spaces) and `test-tag` (without spaces) means the same tag.

serhii-nochevnyi replied:
This will already be the transformation of the tag. In my opinion, it is better to make these changes in a separate ticket.

Commit:
refactored tagUtils:
  tagUtils.isEmpty -> string.isEmptyOrSpaces
  tagUtils -> utils
  tagUtils.removeEmpty -> utils.filterEmptyItems
